ext/liquid_c/vm_assembler.c

```diff
@@ -25,12 +26,23 @@ void vm_assembler_gc_mark(vm_assembler_t *code)
         switch (*ip++) {
```

Comment:
Can we mark the ops that need to be marked and call `liquid_vm_next_instruction` in `vm.c` to move to the next instruction?

Reply:
> get rid of all non-ruby constants from the constants buffer

Where do you plan to put them? It looks like probably interleaved into the code buffer given the implementation of PUSH_INT*, which sounds good to me overall. It does make it more difficult to include medium size non-ruby constants inline. I mentioned in another comment that perhaps we could skip the conversion through RString in some cases, but I also wouldn't want to have strings in the code buffer. (in the OP_LOOKUP_COMMAND we can just use an enum, but I can imagine potentially wanting to skip RString for some drop methods and filter names).

Reply:
Yeah, I wanted to add them to the instructions buffer if they can be used in their serialized form.

I was planning on using an immediate string argument for `OP_WRITE_RAW`, since it seems rare that we would re-use the string and we won't want the deserialization overhead.

For filter, variable and key names, it is actually quite likely that we will re-use the object, so I think a table lookup would make sense.

Longer term, I'm more interested in what is most efficient, where we can re-use code to make things easier.